spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/SparkStreamingPlugin.java

```diff
@@ -73,19 +74,28 @@ public void postProcessModule(Module module) {
 		Properties messageBusProperties = getMessageBusProperties(module, transport);
 		Properties inboundModuleProperties = this.extractConsumerProducerProperties(module)[0];
 		Properties outboundModuleProperties = this.extractConsumerProducerProperties(module)[1];
-		String defaultStorageLevel = env.getProperty(Processor.SPARK_STORAGE_LEVEL_PROP);
+		String defaultStorageLevel = env.getProperty(SparkStreamingSupport.SPARK_STORAGE_LEVEL_PROP);
 		StorageLevel configuredStorageLevel = StorageLevel.fromString(StringUtils.hasText(defaultStorageLevel) ?
 				defaultStorageLevel : DEFAULT_STORAGE_LEVEL);
-		String storageLevelFromModule = module.getProperties().getProperty(Processor.SPARK_STORAGE_LEVEL_MODULE_OPTION);
+		String storageLevelFromModule = module.getProperties().getProperty(SparkStreamingSupport.SPARK_STORAGE_LEVEL_MODULE_OPTION);
 		StorageLevel storageLevel = StringUtils.hasText(storageLevelFromModule) ?
 				StorageLevel.fromString(storageLevelFromModule) : configuredStorageLevel;
 		MessageBusReceiver receiver = null;
 		if (transport.equals("local")) {
-			final Processor sparkModule = module.getComponent(Processor.class);
-			Properties sparkConfigs = SparkStreamingDriverModule.getSparkModuleProperties(sparkModule);
-			String sparkMasterUrl = env.getProperty(Processor.SPARK_MASTER_URL_PROP);
-			if (sparkConfigs != null && StringUtils.hasText(sparkConfigs.getProperty(Processor.SPARK_MASTER_URL_PROP))) {
-				sparkMasterUrl = sparkConfigs.getProperty(Processor.SPARK_MASTER_URL_PROP);
+			SparkStreamingSupport processor;
+			Properties sparkConfigs = null;
+			try {
+				processor = module.getComponent(SparkStreamingSupport.class);
+				if (processor != null) {
```

Comment:
This is the method in question:

````
public <T> T getComponent(Class<T> requiredType) {
   return (this.context.isActive()) ? this.context.getBean(requiredType) : null;
}
````

So as @ilayaperumalg mentions, it can return *null* (whenever the context is not active), or if the context is active but no bean of that type exists, it can throw `NoSuchBeanDefinitionException`.

My question would be if that underlying context would ever be inactive at this point? And if so (thus *null* returned), should we handle that differently, minimally with a different error message?